### DISCUSSION TITLE
libkrb5.a(cc_keyring.o):cc_keyring.c:function get_persistent_real: error: undefined reference to 'keyctl_get_persistent'

### DISCUSSION THREAD

**Discussion by xinda77** [apache/doris#14799] [A - General / Q&A]
centos7.6上编译1.2的时候报错如下 keyutils已经安装啦，并且/usr/include/keyutils.h文件中方法参数列表都对着，怎么会报这个错误，怎样解决呢。

[ 91%] Linking CXX static library libRuntime.a
[100%] Built target Runtime
[100%] Building CXX object src/service/CMakeFiles/doris_be.dir/doris_main.cpp.o
[100%] Linking CXX executable doris_be
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function get_persistent_real: error: undefined reference to 'keyctl_get_persistent'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function session_write_anchor: error: undefined reference to 'keyctl_get_keyring_ID'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function session_write_anchor: error: undefined reference to 'keyctl_get_keyring_ID'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function find_or_create_keyring: error: undefined reference to 'keyctl_search'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function find_or_create_keyring: error: undefined reference to 'add_key'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function find_or_create_keyring: error: undefined reference to 'keyctl_link'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function find_or_create_keyring: error: undefined reference to 'add_key'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function set_primary_name: error: undefined reference to 'add_key'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function get_primary_name: error: undefined reference to 'keyctl_search'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function get_primary_name: error: undefined reference to 'keyctl_search'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function get_primary_name: error: undefined reference to 'keyctl_link'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function get_primary_name: error: undefined reference to 'keyctl_read_alloc'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function unique_keyring: error: undefined reference to 'keyctl_search'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function unique_keyring: error: undefined reference to 'add_key'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function update_keyring_expiration: error: undefined reference to 'keyctl_set_timeout'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_initialize: error: undefined reference to 'keyctl_link'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function clear_cache_keyring: error: undefined reference to 'keyctl_clear'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_destroy: error: undefined reference to 'keyctl_unlink'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_destroy: error: undefined reference to 'keyctl_unlink'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_start_seq_get: error: undefined reference to 'keyctl_read_alloc'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_next_cred: error: undefined reference to 'keyctl_read_alloc'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_get_principal: error: undefined reference to 'keyctl_read_alloc'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_remove_cred: error: undefined reference to 'keyctl_unlink'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_store: error: undefined reference to 'keyctl_set_timeout'
/opt/doris-1.2.0-rc01/thirdparty/installed//lib/libkrb5.a(cc_keyring.o):cc_keyring.c:function krcc_ptcursor_next: error: undefined reference to 'keyctl_describe_alloc'
collect2: error: ld returned 1 exit status
make[2]: *** [src/service/CMakeFiles/doris_be.dir/build.make:199: src/service/doris_be] Error 1
make[1]: *** [CMakeFiles/Makefile2:843: src/service/CMakeFiles/doris_be.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by smallhibiscus** [comment]
u can compline 1.2 with docker image
